xclim/sdba/adjustment.py

```diff
@@ -135,6 +140,44 @@ def _make_dataset(self, **kwargs):
         self.ds = xr.Dataset(data_vars=kwargs)
         self.ds.attrs["adj_params"] = str(self)
 
+    def save_training(
```

Comment:
Should this be a private method ? 
It's mysterious to me that writing to disk provides speed advantages. So I hesitate to make this a public method, and wonder if it should remain hidden (and called automatically behind the scene).

Reply:
I'm not sure on how to automate the calling of this, but if possible that would be great!

Reply:
Reason to why this is useful : dask struggles to produce a task graph when too many operations are needed. When using "time.dayofyear" grouping, there is one "getitem" op per day, so usually 365 doy * 30 years * spatial dims * 2 (ref + hist). This is a humongous amount of operations. My tests showed that dividing the work like this is actually faster than trying to do both adjust and train in one go. The split process was often faster than the task-graph-building of the single process.
Moreover, when trying different adjust for one train, not using  `save_training` makes the training computation re-execute itself each time. (2 / 3)

Reply:
Reasons to why it is public: Because the alternative for a user that wants this option would be:
1) Call a private function `_save_training`
2) Do it by hand with : 
```
DQM.ds.to_netcdf('...')
DQM.__dict__['ds'] = xr.open_dataset('...', **)
```
Which is even worse quant à moi. (3 / 3)

Reply:
Got it. 
Can't you use `.compute()` instead of writing to disk ? 
Could you have a rule deciding whether to force computation or not ? If you have a default value that works well in most cases, it makes this improvement useful for all users, but no one has to care about this until they actually notice a performance issue. 
e.g. something like 

```
force_compute : None, int
  Size of dask operations above which a `compute` on the `train` method will be executed automatically. This is done to avoid the cost of building huge tasks graphs in dask. If None, do not force compute. 
```

Reply:
Hum, I like this `force_compute` idea, will explore.
However, doesn't `compute` put everything in the ram? Our "test" case was tas ad mare usque ad mare, so maybe that's too much for normal ram? In fact, I am following here tip 2 of xarray's http://xarray.pydata.org/en/stable/dask.html#optimization-tips.

Reply:
Yep. I don't know. If it only keeps the training result in memory, then it should not be too bad, right? In other words, if we have `input (DISK) -> chunk streams (RAM) -> train (RAM)`  it should be ok, right ? I mean, we can cap the RAM used by dask, no ? Which will limit the number of chunks it handles at once.

Reply:
Well, with "time­.dayofyear" and, let's say, nquantiles > 30, the training result is actually *larger* than the input (given we use 30 years). I know that this example doesn't make much sense scientifically, but it is a probable use case, no? 
I am currently exploring a way to use `store()` and a hdf5 dataset (on disk) to keep the train lazy while improving reusability of the training results.

Reply:
Indeed. OK !

Reply:
`save_training` and `save_fit` removed.